File: app/applet/extract_keys.py
import re

with open("components/SupervisorDashboard.tsx", "r", encoding="utf-8") as f:
    lines = f.readlines()

for i, line in enumerate(lines):
    if "key=" in line:
        key_str = re.search(r'key=({[^}]*}|"[^"]*"|\'[^\']*\')', line)
        if key_str:
            print(f"Line {i+1}: {key_str.group(1).strip()}")

File: app/applet/extract_keys.ts
import * as fs from 'fs';

const content = fs.readFileSync('components/SupervisorDashboard.tsx', 'utf-8');
const lines = content.split('\n');

lines.forEach((line, i) => {
    if (line.includes('key={') || line.includes('key=')) {
        const match = line.match(/key=({[^}]*}|"[^"]*"|'[^']*')/);
        if (match) {
            console.log(`Line ${i + 1}: ${match[1]}`);
        }
    }
});
